### PR TITLE
Narzedzia #2 homework

### PR DESCRIPTION
**Pull request by mariuszlisowski**
All tasks done, although I had have some difficulties with the last one, i.e. cherry-pick, more closely with a 'hard reset' of 'vimtutor' branch. Thanks to Lukin's hint finally the job was done. Not sure whether it's all correct, but hope so :) But why can't it be automatically merged?

### CODE REVIEW

**Review by Kasperice — APPROVED** [review]
Solution looks good for me

**Review by Marcinati — APPROVED** [review]
Everything is allright.

File: src/functions.cpp [hunk 1]
```diff
@@ -1,9 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"
 #include <iostream>
 #include <limits>
+#include <algorithm>
```
Comment (Marcinati) [inline src/functions.cpp]: Sort headers.

**Review by Marcinati — CHANGES_REQUESTED** [review]
Everything is allright.

File: src/functions.cpp (hunk 1 above)
Comment (mariuszlisowski) [reply src/functions.cpp]: Done. Thank you!

**Review by Marcinati — APPROVED** [review]
Now everything is allright. :)

**Comment by ziobron** [comment]
It can't be merged automatically because I've also added correction on master - missing <algorithm> header.

+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

**Pull request closed without merging**